orttraining/orttraining/python/training/ortmodule/_logger.py

```diff
@@ -76,3 +77,70 @@ class LogColor:
     ENDC = "\033[0m"
     BOLD = "\033[1m"
     UNDERLINE = "\033[4m"
+
+
+class TimeTrackerPhase(IntEnum):
+    EndToEnd = 0
+    EXPORT = 1
+    GRAPH_BUILDER_INIT = 2
+    DETECTION = 3
+    BUILD_GRAPH = 4
+    CREATE_SESSION = 5
+
+    def to_string(self) -> str:
+        if self == TimeTrackerPhase.EndToEnd:
+            return "end to end"
+        if self == TimeTrackerPhase.EXPORT:
+            return "export"
+        elif self == TimeTrackerPhase.GRAPH_BUILDER_INIT:
+            return "graph builder init"
+        elif self == TimeTrackerPhase.DETECTION:
+            return "runtime detection"
+        elif self == TimeTrackerPhase.BUILD_GRAPH:
+            return "building grad"
+        elif self == TimeTrackerPhase.CREATE_SESSION:
+            return "session creation"
+        else:
+            return "invalid"
+
+
+class TimeTracker:
```

Comment:
What is the overhead of adding this tracking?
Can we make it a feature that can be turned off?

Reply:
I think this is just one-time collection, should be trivial, a flag might not needed.